File: src/main/java/com/cardlay/techinterview/cardmanagement/controller/response/CardHolderResponseDto.java
package com.cardlay.techinterview.cardmanagement.controller.response;

import com.cardlay.techinterview.cardmanagement.entity.CardHolder;

public record CardHolderResponseDto(
    Long id,
    String name,
    String email
) {
    public static CardHolderResponseDto assembleFromCardHolder(CardHolder cardHolder) {
        return new CardHolderResponseDto(
            cardHolder.getId(),
            cardHolder.getName(),
            cardHolder.getEmail()
        );
    }
}

File: src/main/kotlin/com/cardlay/techinterview/cardmanagement/controller/response/CardHolderResponseDto.kt
package com.cardlay.techinterview.cardmanagement.controller.response

import com.cardlay.techinterview.cardmanagement.entity.CardHolder

data class CardHolderResponseDto(
    val id: Long,
    val name: String,
    val email: String,
) {
    companion object {
        fun assembleFromCardHolder(cardHolder: CardHolder): CardHolderResponseDto {
            return CardHolderResponseDto(
                cardHolder.id,
                cardHolder.name,
                cardHolder.email
            )
        }
    }
}